ngraph/core/src/op/util/scatter_nd_base.cpp

```diff
@@ -48,47 +52,47 @@ void op::util::ScatterNDBase::validate_and_infer_types()
         this, updates_et == inputs_et, "Updates element type must be the same as inputs");
 
     NODE_VALIDATION_CHECK(this,
-                          indices_shape.rank().is_dynamic() ||
-                              indices_shape.rank().get_length() >= 1,
+                          indices_rank.is_dynamic() || indices_rank.get_length() >= 1,
                           "Indices rank is expected to be at least 1");
 
     NODE_VALIDATION_CHECK(this,
-                          inputs_shape.rank().is_dynamic() || indices_shape.rank().is_dynamic() ||
-                              indices_shape[indices_shape.rank().get_length() - 1].get_length() <=
-                                  inputs_shape.rank().get_length(),
+                          inputs_rank.is_dynamic() || indices_rank.is_dynamic() ||
+                              indices_shape[indices_rank.get_length() - 1].get_length() <=
+                                  inputs_rank.get_length(),
                           "Last dimension of indices can be at most the rank of inputs");
 
-    NODE_VALIDATION_CHECK(
-        this,
-        inputs_shape.rank().is_dynamic() || indices_shape.rank().is_dynamic() ||
-            updates_shape.rank().is_dynamic() ||
-            updates_shape.rank().get_length() ==
-                indices_shape.rank().get_length() + inputs_shape.rank().get_length() -
-                    indices_shape[indices_shape.rank().get_length() - 1].get_length() - 1,
-        "Rank of updates must be rank of inputs + rank of indices - last dimension of indices "
-        "- 1");
-
-    bool compatible = true;
-    if (inputs_shape.is_static() && indices_shape.is_static() && updates_shape.is_static())
+    if (inputs_rank.is_static() && indices_rank.is_static() && updates_rank.is_static())
     {
-        size_t indices_rank = indices_shape.rank().get_length();
-        size_t updates_rank = updates_shape.rank().get_length();
-        for (size_t i = 0; i < indices_rank - 1; i++)
-        {
-            compatible = compatible && updates_shape[i].same_scheme(indices_shape[i]);
-            NODE_VALIDATION_CHECK(
-                this,
-                compatible,
-                "updates_shape[0:indices_rank-1] shape must be indices_shape[:-1]");
-        }
-        size_t j = indices_shape[indices_rank - 1].get_length();
-        for (size_t i = indices_rank - 1; i < updates_rank; i++, j++)
+        auto expected_updates_rank = indices_rank.get_length() + inputs_rank.get_length() -
+                                     indices_shape[indices_rank.get_length() - 1].get_length() - 1;
+        // If expected updates rank is 0D it also can be a tensor with one element
+        NODE_VALIDATION_CHECK(
+            this,
+            updates_rank.get_length() == expected_updates_rank || expected_updates_rank == 0,
```

Comment:
when `expected_updates_rank == 0` you also need to check `update_shape`

Reply:
We check rank here, rank can't be checked for this case.